### PR TITLE
Add support for blend of numeric and symbolic parameters

### PR DESCRIPTION
**Pull request by SamDuffield**
Added support for a combination of numeric and symbolic parameters (essentially a bug fix)

Added conversion of non-parameterised gates not found in `qujax.gates` in the symbolic implementation. This is not immediately feasible in the non-symbolic implementation since we look in `qujax.gates` to determine parameterised vs non-parameterised gates, have added a raise TypeError in this case.

#42 #40

### CODE REVIEW

**Review by kplekhanov — APPROVED** [review]
Looks good but may be nice to actually "compile" qujax gates by making numerical pytket parameters constants and leaving symbolic pytket parameters as actual qujax circuit parameters

**Review by kplekhanov — APPROVED** [review]

**Review by kplekhanov — DISMISSED** [review]

File: pytket/extensions/qujax/qujax_convert.py [hunk 1]
```diff
@@ -23,6 +23,10 @@
 from pytket import Qubit, Circuit  # type: ignore
 
 
+class NamedArray(jnp.DeviceArray):
+    __name__: str
```
Comment (cqc-melf) [inline pytket/extensions/qujax/qujax_convert.py:27]: Do you want to add an additional constructor here? What is the default value?

Comment (SamDuffield) [reply pytket/extensions/qujax/qujax_convert.py:27]: No lol this was an idea of mine to enable cleaner printing of the "qujax args" but in the end I settled on the simpler approach of just leaving the converted gates as jnp.DeviceArray instances

File: pytket/extensions/qujax/qujax_convert.py [hunk 2]
```diff
@@ -91,6 +91,10 @@ def tk_to_qujax_args(
         gate_name = c.op.type.name
         if gate_name == "Barrier":
             continue
+        if gate_name == "Measure":
```
[the remark this answers is not in the document]
Comment (SamDuffield) [reply pytket/extensions/qujax/qujax_convert.py:94]: Done

File: tests/test_tket_symbolic.py [hunk 3]
```diff
@@ -126,6 +126,28 @@ def test_CZ_qrev() -> None:
     _test_circuit(circuit, symbols, True)
 
 
+def test_symbolic_numeric_blend() -> None:
```
[the remark this answers is not in the document]
Comment (SamDuffield) [reply tests/test_tket_symbolic.py:129]: Done

File: docs/changelog.rst [hunk 4]
```diff
@@ -1,6 +1,11 @@
 Changelog
 ~~~~~~~~~
 
+0.5.1 (September 2022)
```
Comment (cqc-melf) [inline docs/changelog.rst:4]:
```suggestion
0.6.0 (September 2022)
```

File: _metadata.py [hunk 5]
```diff
@@ -1,2 +1,2 @@
-__extension_version__ = "0.5.0"
+__extension_version__ = "0.5.1"
```
Comment (cqc-melf) [inline _metadata.py:1]:
```suggestion
__extension_version__ = "0.6.0"
```

Comment (cqc-melf) [reply _metadata.py:1]: Would go with 0.6.0 because we will update the pytket version, too.

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**